mevislab.github.io/content/tutorials/basicmechanisms/macromodules.md

```diff
@@ -15,16 +15,104 @@ menu:
 
 # Example 2: Macro modules {#TutorialChapter6}
 
-## Macro modules and Module Interactions via User Interface and Python Scripting
+## Macro Module 
+A macro module can be used to develop your own functionality in MeVisLab. You have two main options for how to use it:
 
-MeVisLab provides different types of modules, which can be distinguished by their color. The brown modules are called Macro modules. Macro modules condense a whole network into one module. You can open the internal network by pressing the middle mouse button {{< mousebutton "middle" >}} or via right mouse click {{< mousebutton "right" >}} and select {{< menuitem "Help" "Show Internal Network" >}}. Macro modules provide the possibility to create customized user interfaces and Python interactions.
+* **With Internal Networks**: Use a macro module to reuse a network of modules. For example, if you build a network that applies a specific image filter and you want to use this setup in multiple projects, you can wrap the entire network into a single macro module. This way, you don’t need to manually reconnect all the individual modules each time — you just use your macro module.You can see in the  left side of the image is the Internal network. You can see an example in [Basic Mechanics of MeVisLab (Example: Building a Contour Filter)](/tutorials/basicmechanisms#TutorialMacroModules).
```

Comment:
.You can see in the left side of the image is the Internal network. ->
. You can see the internal network at the left side of the image.
- which image this refers to?